docs/ProviderDesign.md

```diff
@@ -0,0 +1,79 @@
+# Provider's view of the FLOCX
+
+![ProviderDesign](../images/ProviderService-Marketplace.png)
+
+
+
+
+## Multi-tenant IRONIC:
+Ironic manages multiple pools of hardware each owned by a separate user/group/organization.
+
+
+## Provider Service:
+A centralized service per OpenStack deployment that manages offers and contracts 
+on behalf of all its tenants of the OpenStack.
+
+## Keystone:
+Provides authentication and authorization services.
+
+
+## Workflow:
+
+Following steps describe the process of offering physical hardware in the market by individual owners
+and receiving contracts from the marketplace.
+
+
+
+1. Hardware pool owners decide which nodes to offer in the marketplace and submit offers to the provider service
+2. Provider service confirms owner-hardware relationship before accepting offers. 
+3. For each accepted offer it fetches the hardware configuration from Ironic
+4. These offers are published to the marketplace as soon as possible. 
+5. Receives contracts from the marketplace.
```

Comment:
When we talked about this, I think it was decided that the provider *pulled* contracts from the marketplace. Is that still correct?

Reply:
This part is not clear in my understanding. Once a contract is made, who executes it, provider service or the hardware owner or some other OpenStack service? Because setting up the contract requires providing necessary networking also.

Reply:
The provider service has to be able to manage contracts on its side; this way it can still expire contracts if it gets disconnected from the marketplace